ingester/src/deferred_load.rs

```diff
@@ -68,6 +73,85 @@ where
     }
 }
 
+impl<T> Display for DeferredLoad<T>
+where
+    T: Display,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self.value.lock().as_ref().unwrap() {
+            State::Unresolved(_) | State::Loading(_) => f.write_str(UNRESOLVED_DISPLAY_STRING),
+            State::Resolved(v) => v.fmt(f),
+        }
+    }
+}
+
+impl<T> DeferredLoad<T> {
+    /// Provide a hint to the [`DeferredLoad`] that the value will be used soon.
+    ///
+    /// This allows the value to be materialised in the background, in parallel
+    /// while the caller is executing code that will eventually call
+    /// [`Self::get()`].
+    pub(crate) fn prefetch_now(&self) {
+        let mut state = self.value.lock();
+
+        // If the value has already resolved, this call is a NOP.
+        if let Some(State::Resolved(_)) = &*state {
+            return;
+        }
+
+        // Potentially transition the state, discarding the waker.
+        let (_, new_state) = self.get_load_waker(state.take().unwrap());
```

Comment:
Nit, but it takes a few brain cells to go from "discarding the waker" to realizing that the underscore is the waker, could we save those brain cells by doing this instead?

```suggestion
        // Potentially transition the state, discarding the waker.
        let (_waker, new_state) = self.get_load_waker(state.take().unwrap());
```

Reply:
Great idea - I sure need to hang onto as many brain cells as possible.